### PR TITLE
Raise exception when no recipe is provided

### PR DESCRIPTION
**Pull request by tomchop**

### CODE REVIEW

File: dftimewolf/cli/dftimewolf_recipes.py [hunk 1]
```diff
@@ -164,6 +164,11 @@ def ParseArguments(self, arguments):
     self._AddRecipeOptions(argument_parser)
 
     self._command_line_options = argument_parser.parse_args(arguments)
+
+    if not hasattr(self._command_line_options, 'recipe'):
```
Comment (joachimmetz) [inline dftimewolf/cli/dftimewolf_recipes.py]: what if recipe is set but None? Maybe:

```
if not getattr(self._command_line_options, 'recipe', None):
```

File: dftimewolf/lib/errors.py [hunk 2]
```diff
@@ -11,3 +11,6 @@ def __init__(self, message=None):
 
 class RecipeParseError(DFTimewolfError):
   """Error when parsing a recipe."""
+
```
Comment (joachimmetz) [inline dftimewolf/lib/errors.py]: `+1` white line

File: dftimewolf/cli/dftimewolf_recipes.py [hunk 3]
```diff
@@ -221,8 +224,10 @@ def Main():
     print('{0!s}'.format(exception))
     return False
 
-  if not tool.ParseArguments(sys.argv[1:]):
-    # TODO: print errors if this fails.
+  try:
+    tool.ParseArguments(sys.argv[1:])
+  except errors.CommandLineParseError as exception:
+    sys.stderr.write(str(exception))
```
Comment (joachimmetz) [inline dftimewolf/cli/dftimewolf_recipes.py]: Isn't sys.stderr.write() expecting bytes (on Python 3)?

File: dftimewolf/cli/dftimewolf_recipes.py (hunk 1 above)
Comment (tomchop) [reply dftimewolf/cli/dftimewolf_recipes.py]: Interesting, I was pretty sure the default return value for `getattr` if no third argument was provided was already None. I should revisit other codebases 😅. Thanks!

File: dftimewolf/cli/dftimewolf_recipes.py (hunk 3 above)
Comment (tomchop) [reply dftimewolf/cli/dftimewolf_recipes.py]: Apparently not, write seems to expect `str`

```
$ python3
Python 3.7.3 (default, Mar 27 2019, 09:23:15)
[Clang 10.0.1 (clang-1001.0.46.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.stderr.write(b'asd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: write() argument must be str, not bytes
```

**Comment by tomchop** [comment]
Done, PTAL

File: dftimewolf/cli/dftimewolf_recipes.py (hunk 1 above)
Comment (joachimmetz) [reply dftimewolf/cli/dftimewolf_recipes.py]: It likely is, but I prefer it to be specific

Comment (tomchop) [reply dftimewolf/cli/dftimewolf_recipes.py]: Ack, thanks for the clarification.

**Pull request merged**